src/main/java/mekanism/common/tile/base/TileEntityMekanism.java

```diff
@@ -431,8 +438,46 @@ public void addComponent(ITileComponent component) {
     }
 
     @Nonnull
+    @Override
+    @SuppressWarnings("ConstantConditions")
     public Component getName() {
-        return TextComponentUtil.translate(Util.makeDescriptionId("container", getBlockType().getRegistryName()));
+        final var defaultName = TextComponentUtil.build(getBlockType());
+        if (isNameable()) {
+            return hasCustomName() ? getCustomName() : defaultName;
+        }
+        return defaultName;
```

Comment:
Given getCustomName returns null if it isn't nameable, this can stay as it was before `return hasCustomName() ? getCustomName() : TextComponentUtil.build(getBlockType());`